draft-ietf-httpbis-digest-headers.md

```diff
@@ -77,442 +76,413 @@ normative:
 SEMANTICS}}).
-Both mechanisms operate independent of transport integrity, offering
-the potential to detect programming errors and corruption of data in flight or
-at rest.
-They can be used across multiple hops in order to provide end-to-end
-integrity guarantees, which can aid fault diagnosis when resources are
-transferred across hops and system boundaries.
-Finally, they can be used to
-validate integrity when reconstructing a resource fetched using different HTTP
-connections.
-
-This document obsoletes [RFC3230].
+Second, representation data integrity, which acts on representation data ({{Section 3.2
+of SEMANTICS}}). This supports more advanced use cases such as validating the
+integrity of a resource the was reconstructed from parts retrieved using
+multiple requests or connections.
+
+This document obsoletes RFC 3230 and therefore the Digest and Wants-Digest HTTP
+fields; see {{obsolete-3230}}.
+
 
 ## Document Structure
 
 This document is structured as follows:
 
-- {{digest}} describes concepts related to representation digests. It also
-  defines the Digest request and response header and trailer field,
+- {{representation-digest}} defines the Representation-Digest request and response header and trailer field,
 - {{content-digest}} defines the Content-Digest request and response header and trailer field,
-- {{want-fields}} defines the Want-Digest and Want-Content-Digest request and response header and
+- {{want-fields}} defines the Want-Representation-Digest and Want-Content-Digest request and response header and
   trailer field,
-- {{algorithms}} describes algorithms and their relation to Digest,
+- {{algorithms}} describes algorithms and their relation to the fields defined in this document,
 - {{state-changing-requests}} details computing representation digests,
 - {{examples-unsolicited}} and {{examples-solicited}} provide examples of using
-  Digest and Want-Digest.
+  Representation-Digest and Want-Representation-Digest.
 
 ## Concept Overview
 
-This document defines the `Digest` request and response header and trailer
-field; see {{digest}}. At a high level, the value contains a checksum, computed
-over `selected representation data` ({{Section 3.2 of
-SEMANTICS}}), that the recipient can use to
-validate integrity. Basing `Digest` on the selected representation makes it
-straightforward to apply it to use-cases where the transferred data requires
-some sort of manipulation to be considered a representation or conveys a
-partial representation of a resource, such as Range Requests (see {{Section
-14.2 of SEMANTICS}}).
-
-To support use-cases where a simple checksum of the content bytes is required,
-this document introduces the `Content-Digest` request and response header and trailer field; see
+The HTTP fields defined in this document can be used for HTTP integrity. Senders
+choose a hashing algorithm and calculate a checksum from an input related to the
```

Comment:
```suggestion
choose a hashing algorithm and calculate a digest from an input related to the
```

"checksum" has a particular meaning, where I *think* digest is more generic.

Reply:
@martinthomson is `hash` ok?